irods_specific_implementation/irods_rest_logical_path_delete_api_implementation.h

```diff
@@ -0,0 +1,88 @@
+#include "irods_rest_api_base.h"
+
+#include "constants.hpp"
+#include <irods/filesystem.hpp>
+#include <irods/rodsClient.h>
+#include <irods/irods_query.hpp>
+
+#include <pistache/http_headers.h>
+#include <pistache/optional.h>
+#include <pistache/router.h>
+
+#include <nlohmann/json.hpp>
+
+using json = nlohmann::json;
+
+namespace irods::rest
+{
+    // this is contractually tied directly to the api implementation
+    const std::string service_name{"irods_rest_cpp_logical_path_delete_server"};
+
+    class logical_path_delete : public api_base
+    {
+    public:
+        logical_path_delete()
+            : api_base{service_name}
+        {
+            info("Endpoint initialized.");
+        }
+
+        std::tuple<Pistache::Http::Code, std::string>
+        operator()(const Pistache::Rest::Request& _request,
+                   Pistache::Http::ResponseWriter& _response)
+        {
+            namespace fs = irods::experimental::filesystem;
+
+            try {
+
+                auto _logical_path = _request.query().get("logical-path").get();
+                auto _no_trash     = _request.query().get("no-trash").getOrElse("0");
+                auto _verbose      = _request.query().get("verbose").getOrElse("0");
+                auto _progress     = _request.query().get("progress").getOrElse("0");
+                auto _recursive    = _request.query().get("recursive").getOrElse("0");
+                auto _unregister   = _request.query().get("unregister").getOrElse("0");
+
+                bool no_trash      = http_bool(_no_trash);
+                bool verbose       = http_bool(_verbose);
+                bool progress      = http_bool(_progress);
+                bool recursive     = http_bool(_recursive);
+                bool unregister    = http_bool(_unregister);
+
+                auto conn = get_connection(_request.headers().getRaw("authorization").value());
+
+                const fs::path logical_path ( _logical_path.c_str() );
+
+                fs::extended_remove_options opts {
+                    .no_trash      = no_trash,
+                    .verbose       = verbose,
+                    .progress      = progress,
+                    .recursive     = recursive,
+                    .unregister    = unregister
+                };
+
+                bool status  = fs::NAMESPACE_IMPL::remove(*conn(), logical_path, opts);
+
+                nlohmann::json json_obj = {
+                    {"op_status", status ? op_success : op_failure}
```

Comment:
Does `status` always mean success or failure?
What does success/failure mean in the context of `fs::client::remove(...)`?
Can the meaning of success/failure change depending on whether we're dealing with a data object or collection?

*These questions apply to irods_rest_cpp_logical_path_rename_server as well.*

Reply:
Just thinking out loud: the return value of ```fs::remove``` comes from the function ```remove_impl```, which is bool. ```remove_impl``` returns bool in the case where the path doesn't point to an existing data object or collection, so ```false``` must mean error. Moreover, we can't really what's gone wrong if the function returns false: either it worked (true) or it didn't (false). However, some paths through those functions throw errors which will be reported by the catch blocks in the bottom. 

Does this hold water?

Reply:
Sounds good to me :)